backend/capellacollab/health/routes.py

```diff
@@ -46,31 +47,49 @@ def general_status(db: orm.Session = fastapi.Depends(database.get_db)):
         )
     ],
 )
-def model_status(
+async def model_status(
     db: orm.Session = fastapi.Depends(database.get_db),
     logger: logging.LoggerAdapter = fastapi.Depends(
         core_logging.get_request_logger
     ),
 ):
+    toolmodels = toolmodels_crud.get_models(db)
+    tasks_per_model = {
+        model.id: {
+            "primary_git_repository_status": asyncio.create_task(
+                git_validation.check_primary_git_repository(db, model, logger)
+            ),
+            "model_badge_status": asyncio.create_task(
+                modelbadge_validation.check_model_badge_health(
+                    db, model, logger
+                )
+            ),
+            "diagram_cache_status": asyncio.create_task(
+                diagrams_validation.check_diagram_cache_health(
+                    db, model, logger
+                )
+            ),
+        }
+        for model in toolmodels
+    }
+
+    results_per_model = {}
+    for model_id, models_tasks in tasks_per_model.items():
+        results_per_model[model_id] = {}
+        for task_name, task in models_tasks.items():
+            results_per_model[model_id][task_name] = await task
```

Comment:
This bit is not needed if you follow the next suggestion.

Reply:
Nice cleanup.